src/library/mod.rs

```diff
@@ -2,14 +2,29 @@ mod http;
 mod sqlite;
 mod toc;
 
-pub struct Library {}
+pub struct Library {
+    pub file_name: Option<String>,
+    pub markdown_content: Option<String>,
+}
 
 impl ftd::p2::Library for Library {
     fn get(&self, name: &str) -> Option<String> {
         if name == "fpm" {
-            return Some(fpm::fpm_ftd().to_string());
-        }
-        if let Ok(v) = std::fs::read_to_string(format!("./{}.ftd", name)) {
+            // append the vars in the string
+            let fpm_base = fpm::fpm_ftd().to_string();
+            let fpm_base = if self.file_name.is_some() && self.markdown_content.is_some() {
+                format!(
+                    "{}\n-- string markdown-filename:\n{} \n-- string markdown-content:\n{}",
```

Comment:
Rust supports multiline strings and also there is [`indoc`](https://docs.rs/indoc/1.0.3/indoc/index.html), avoid `\n` in Rust strings so its easy to read.

Also avoid `{}` and use [named parameters to `format!()`](https://www.oreilly.com/library/view/learning-rust/9781785884306/61dd6807-b2b7-4610-b787-e761f878c073.xhtml) etc.